bioconda_utils/lint_functions.py

```diff
@@ -0,0 +1,186 @@
+    if build_section:
+        build_number = int(build_section.get('number', 0))
+    return df[
+        (df.name == name) &
+        (df.version == version) &
+        (df.build_number == build_number)
+    ]
+
+
+def in_other_channels(recipe, meta, df):
+    """
+    Does the package exist in any other non-bioconda channels?
+    """
+    results = _subset_df(recipe, meta, df)
+    channels = set(results.channel).difference(['bioconda'])
+    if len(channels):
+        return {
+            'exists_in_channels': channels,
+            'fix': 'consider deprecating',
+        }
+
+
+def already_in_bioconda(recipe, meta, df):
+    """
+    Does the package exist in bioconda?
+    """
+    results = _subset_df(recipe, meta, df)
+    channels = set(results.channel)
+    if 'bioconda' in channels:
+        return {
+            'already_in_bioconda': True,
+            'fix': 'bump version or build number'
+        }
+
+
+def missing_home(recipe, meta, df):
+    try:
+        meta['about']['home']
+    except KeyError:
+        return {
+            'missing_home': True,
+            'fix': 'add about:home',
+        }
+
+
+def missing_summary(recipe, meta, df):
+    try:
+        meta['about']['summary']
+    except KeyError:
+        return {
+            'missing_summary': True,
+            'fix': 'add about:summary',
+        }
+
+
+def missing_license(recipe, meta, df):
+    try:
+        meta['about']['license']
+    except KeyError:
+        return {
+            'missing_license': True,
+            'fix': 'add about:license'
+        }
+
+
+def missing_tests(recipe, meta, df):
+    test_files = ['run_test.py', 'run_test.sh', 'run_test.pl']
+    if 'test' not in meta:
+        if not any([os.path.exists(os.path.join(recipe, f)) for f in
+                    test_files]):
+            return {
+                'no_tests': True,
+                'fix': 'add basic tests',
+            }
+
+
+def missing_hash(recipe, meta, df):
+    # could be a meta-package if no source section or if None
+    try:
+        src = meta['source']
+        if src is None:
+            return
+    except KeyError:
+        return
+
+    if not any(
+        (
+            'md5' in src,
+            'sha1' in src,
+            'sha256' in src
+        )
+    ):
+        return {
+            'missing_hash': True,
+            'fix': 'add md5, sha1, or sha256 hash to "source" section',
+        }
+
+
+def uses_git_url(recipe, meta, df):
+    try:
+        src = meta.get('source', {})
+        if src is None:
+            # metapackage?
+            return
+
+        if 'git_url' in src:
+            return {
+                'uses_git_url': True,
+                'fix': 'use tarballs whenever possible',
+            }
+    except KeyError:
+        return
+
+
+def _get_deps(meta):
+    reqs = meta.get('requirements')
+    if reqs is None:
+        return []
+    return [
+        i for i in
+        [reqs.get('build'), reqs.get('run'), reqs.get('test')]
```

Comment:
In my tests this gets the dependencies as listed in the meta.yaml, so version string is not included and the `uses_perl_threaded` test works. In fact many of our existing recipes fail the test because they have perl-threaded (but I think the perl team already knows that)